src/darwin/Framework/CHIP/MTRDeviceAttestationDelegate.h

```diff
@@ -21,12 +21,36 @@ NS_ASSUME_NONNULL_BEGIN
 
 @class MTRDeviceController;
 
+@interface MTRDeviceAttestationDeviceInfo : NSObject
+@property (nonatomic, readonly) NSData * dacCertificate;
+@property (nonatomic, readonly) NSData * dacPAICertificate;
+@property (nonatomic, readonly) NSData * certificateDeclaration;
+@end
+
 /**
  * The protocol definition for the MTRDeviceAttestationDelegate
  *
  * All delegate methods will be called on the callers queue.
  */
 @protocol MTRDeviceAttestationDelegate <NSObject>
+@optional
+/**
+ * Notify the delegate when device attestation completed with device info for additional verification. If
+ * this callback is implemented, continueCommissioningDevice on MTRDeviceController is expected
+ * to be called if commisioning should continue.
```

Comment:
Should probably document how this interacts with the failedForDevice: signature.  What will happen is this one will be called if implemented, right?  Otherwise the other will be, but only if attestation failed?

Reply:
I can document this more clearly - only one of these should be implemented, as only one will be called.

I would actually like to entirely remove the failed callback at least for Darwin, and always call the completed callback.

Reply:
I've documented this better. Please have a look!